influxql/ast.go

```diff
@@ -1748,6 +1767,46 @@ func (s *SelectStatement) validateDerivative() error {
 
 	return nil
 }
+func (s *SelectStatement) validateElapsed() error {
```

Comment:
We're currently trying to remove the `derivative()` validation code since I found it doesn't actually get used. The primary restriction in here is the limitation on how many fields can be used. Most of the validation should be done in `validateAggregates()`.